lib/Service/Provisioning/Manager.php

```diff
@@ -155,17 +153,12 @@ private function updateAccount(IUser $user, MailAccount $account, Config $config
 		return $account;
 	}
 
-	public function deprovision(): void {
-		$this->mailAccountMapper->deleteProvisionedAccounts();
-
-		$config = $this->configMapper->load();
-		if ($config !== null) {
-			$config->setActive(false);
-			$this->configMapper->save($config);
-		}
+	public function deprovision(Provisioning $provisioning): void {
+		$this->mailAccountMapper->deleteProvisionedAccounts($provisioning->getId());
+		$this->provisioningMapper->setActive(false, $provisioning);
 	}
 
-	public function importConfig(array $data): Config {
+	public function importConfig(array $data): Provisioning {
```

Comment:
the import is called from ``\OCA\Mail\Migration\MigrateProvisioningConfig::run`` and I think at this point we could even drop this old repair step as most instances will have upgraded.

Reply:
Deleted!